### ISSUE TITLE
Windows 10 version bug

### ISSUE THREAD

**Issue by BobbyJ72** [bitwarden/desktop#539]
## Describe the Bug

The two versions are no longer consistant.   While the android version will allow you to get to passwords even if your connection to sever vault is down.   The PC version will not allow you to access the vault to see passwords if vault access is down.   Was locked out of a combo out of range of server on laptop and bitwarden PC would not allow access.   

## Steps To Reproduce

Try to get to your vault from windows 10 (and have no internet connection) you can't but you can on a phone.   Please fix allow local cached vault for times when net is down.  

4. Click on '...'

## Environment

- Operating system: [e.g. Windows 10, Mac OS Catalina]
- Build Version (go to "Settings" → "About"): [e.g. 1.16.10]

**Comment by Greenderella** [comment]
Hi @BobbyJ72 

I'm unable to reproduce this issue: https://drive.google.com/file/d/1AjUDXfPL9wzIARrgdBR5tvwxEVBpaBJH/view?usp=sharing

Will you first try uninstalling the app, removing any leftover data, resetting your device and re-installing? Does this issue still persist?

Removal of the leftover data is important. New installation may hold on to old installation files and cause the same issues.
Guide to the leftover data: https://help.bitwarden.com/article/where-is-data-stored-computer/

**Comment by BobbyJ72** [comment]
You can't reproduce it because you were already logged in then just locked
it.   I am talking about booting your pc with no wifi or internet then
loading bitwarden and trying to log in.

Also I am running my own bitwarden server that I access via VPN at office.
that is my server url.  Since its VPN im on the network so no need for
https (exposing it to the internet).    The issue is if I am not on VPN and
attempt to log in it errors.

[image: image.png]

The error is show below.    On my phone I do the exact same thing, logging
in via VPN but I don't get the failed to fetch error message.  It accesses
the passwords stored in the phone.   For some reason I can't do the same on
the PC version.
[image: image.png]

**Comment by vachan-maker** [comment]
I turned of my wifi and tried to connect to my vault and got a **failed to fetch error**  . Before this version, I am pretty sure I could access the vault without internet.

**Comment by davidg246819** [comment]
@BobbyJ72 what server version are you using? I have the same issue after updating the Windows desktop app today. I run bitwarden_rs. Android app seems to work fine but I suppose is returning cached records?

**Comment by BobbyJ72** [comment]
What ever was latest 3 months ago.    Docker 19.04.6 running on Ubuntu
18.04.3 it pulled the latest version on the installation but I don't know
what version it was.

**Comment by BobbyJ72** [comment]
I tried it on two different computers from different locations and get the
same fail to fetch error.

But a buddy of mine with his own bitwarden server can do it without that
error.   I'm at a loss as to why.

**Comment by BobbyJ72** [comment]
I did all you asked and still have the same issue.   Bit warden client had
like 2-3 updates in the past 2 days none of those made any difference.   I
tried it on a PC on a different site and can log in (if vpn is connected)
but if I close bitwarden and turn off the VPN it will busy bee for a little
bit as if its still trying to connect to the server.  Then it gives up with
a fetch error or wrong name/password error.    Then if I turn VPN back on
it connects fine.

It appears its still looking for the ip to the server if it does not find
it I would assume it would check for a LOCAL version of vault and load
it.   For some reason it does not.

I know it works on a buddies and he set up his server around the same time
with the same commands as me.   Could it be some kind of server issue
causing it not to properly create a local file?     When I click syn i get
the green sync OK pop up.   And on my android it works perfectly.    I am
at a loss, but I know I am not the other person with this issue.

Maybe have a button that says OFFILINE MODE which forces it to use the
local file.

**Comment by cscharf** [comment]
I would assume there are some OSs (like OSX for example) where rebooting the machine will suspend and maintain state of open/running applications so when it starts back up again, it's the OS maintaining the application state for you (not the app itself) and therefore you are able to unlock the vault w/o connectivity to the server (local or cloud). However, if someone were you log out of their vault regardless of a reboot or not, and then try to log in w/o connectivity to the server, it would and should never work, logging out is designed to completely destroy your local vault or in-memory constructs therefore you absolutely _must_ connect to the server when logging back into your vault to pull/sync your vault data.

If your buddy is able to have this work when he reboots on his linux distro or other OS, you can't really compare behavior due to the way application state is managed by various OSs.

In your case, in Windows, if closing the Bitwarden application, you're likely also logging out and are therefore required to log back in (vs. the application simply running in the background).

**Comment by BobbyJ72** [comment]
We both run windows 10 for our work.

We use a linux VM on a server to run bitwarden server.

On my phone (Android) i can have zero connectivity) and get to my vault
every time.
On my windows 10 PC i can't get to my vault no matter what I do.     I have
not be on VPN and connected to the linux server.  Yet my buddy with same
windows 10 can pull the plug on his computer (no internet) load up bit
warden and get to his local vault to see pass words.

So we have same set ups but one can access data offline and other can't.

My android saves a local vault why can't windows 10?

**Comment by cscharf** [comment]
If your buddy is willing, I would suggest you follow steps side-by-side on a screen share perhaps to ensure you're both doing the exact same thing. There are also always a multitude of ways to configure Windows 10, other software, etc. which could potentially interfere with an operating environment. Obviously offline access works for a locked vault (not logged out vault) in most systems, as designed. So if this is not working for you there may be something else at play, however it is apparently still working as designed if your friend is able to use it in this manner.

We do also already have a community requested feature for "offline vault management" for true offline editing of items: https://community.bitwarden.com/t/offline-management-of-writeable-vault-items/107/38

There is also several mentions of issues where users who are offline, not connected via VPN or otherwise do not have connectivity to their self-hosted BW server (or our cloud servers) are not able to log in; however they are able to unlock their vault (as expected). Are you perhaps logging out vs. simply locking the vault. What is your timeout/close behavior (lock or log out)?

**Comment by BobbyJ72** [comment]
My timeout behavior is locked.

But I just tested, it seems I have to log in to get to my vault.   If I
click locked and try to get back in I get an error message.  So I have to
log out then log in to get to my data

I am typing the correct password but its LOCKED, the error is not correct
unless there is a different master pass for locked vs log in status.

[image: image.png]

**Comment by cscharf** [comment]
> unless there is a different master pass for locked vs log in status.

No, it would be the same master password regardless of locked vs. logged out status.

> What ever was latest 3 months ago.    Docker 19.04.6 running on Ubuntu 18.04.3 it pulled the latest version on the installation but I don't know what version it was.

@BobbyJ72 , have you tried to update to the latest server version?

**Comment by BobbyJ72** [comment]
I feel the product might be ok for people that use the official web
interface but the self hosted is a work in progress.
I am unsure as to have to even back up the vault on the server and do not
trust my local vault. since I can't access it.

What I've come up with is to use SCP to remote to the server over vpn and
manually copy the bw-data folder to an offsite NAS.

Another limitation is users have to use a email to make a vault.   Say I
want a community vault for the office where trusted employees can access
the office account passwords for say venders.    Lasspass can do it but bit
warden cannot.

I may try to reinstall the entire VM at some point but not until I see the
issues have been rectified.

[image: image.png]

**Comment by cscharf** [comment]
@BobbyJ72 , I hate to do this as I want to ensure you get the support that you need, but our vault simply cannot support login "offline" when you're logged out (this has never worked and likely never will). Locked vault, offline, no problem. Logged out, that's another story.

If you are having issues with backing up your vault, I would highly suggest you post in our [Bitwarden Community Forums](https://community.bitwarden.com) where there are a lot of knowledgeable folks, including our customer success team, who can help you with that. We have many world-class enterprises who use our self-hosted solution without issue and have operationalized it as a mission-critical part of their business operations (which one could imagine includes secure backups).

> Another limitation is users have to use a email to make a vault. Say I want a community vault for the office where trusted employees can access the office account passwords for say venders.  Lasspass can do it but bit warden [sic] cannot.

Having to use email to create a vault does sound like a potential limitation in that scenario where you would want anonymous (but trusted employee?) users accessing user secrets without an email address, if you would like to create a feature request I would highly recommend you do so on the forums as it has a section for submitting, voting for, and discussing product [feature requests](https://community.bitwarden.com/c/feature-requests) like this one; although we do offer Organizations with the ability to share secrets including groups, permissions, etc. providing what seems like **exactly** the use-case you've described here (and I'm still not clear on how email plays a role there).

This issue seems to be morphing in too many directions and we are not able to reproduce the original issue reported; other than when the Bitwarden application is closed or logged out (which is by design).

**Comment by BobbyJ72** [comment]
On my bitwarden I can't lock it and log back in.    It does not work, not
matter if I'm offline or online.   So my only option is to LOGOFF and Logon
to the vault.    (this is windows 10 client, android is fine)
But my phone on that same server vault has zero issues locking and
accessing the vault off or online.

I think it must have to do with the server installation as I can duplicate
it on any windows 10 machine, i've tried it on 5 now with the same results.
So to me this points to the server or windows client.   As you said you
can't duplicate it so it would be the server.

Thanks for your time and information.
